src/UI/WB.UI.Designer/Controllers/Api/Designer/AuthorizeOrAnonymousQuestionnaireAttribute.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using WB.Core.BoundedContexts.Designer.Views.Questionnaire.ChangeHistory;
+using WB.Core.BoundedContexts.Designer.Views.Questionnaire.Edit;
+
+namespace WB.UI.Designer.Controllers.Api.Designer
```

Comment:
Could you please add at lease one test that verifies that questionnaire not actively shared is not returned?

Reply:
added